src/middlewared/middlewared/plugins/pool.py

```diff
@@ -300,6 +304,376 @@ def pool_extend(self, pool):
+            'autoexpand': 'on',
+        }
+
+        fsoptions = {
+            'compression': 'lz4',
+            'aclmode': 'passthrough',
+            'aclinherit': 'passthrough',
+            'mountpoint': f'/{data["name"]}',
+        }
+
+        dedup = data.get('deduplication')
+        if dedup:
+            fsoptions['dedup'] = dedup.lower()
+
+        cachefile_dir = os.path.dirname(ZPOOL_CACHE_FILE)
+        if not os.path.isdir(cachefile_dir):
+            os.makedirs(cachefile_dir)
+
+        job.set_progress(90, 'Creating ZFS Pool')
+        z_pool = await self.middleware.call('zfs.pool.create', {
+            'name': data['name'],
+            'vdevs': vdevs,
+            'options': options,
+            'fsoptions': fsoptions,
+        })
+
+        job.set_progress(95, 'Setting pool options')
+        pool_id = None
+        try:
+            # Inherit mountpoint after create because we set mountpoint on creation
+            # making it a "local" source.
+            await self.middleware.call('zfs.dataset.update', data['name'], {
+                'properties': {
+                    'mountpoint': {'source': 'INHERIT'},
+                },
+            })
+            await self.middleware.call('zfs.dataset.mount', data['name'])
+
+            pool = {
+                'name': data['name'],
+                'guid': z_pool['guid'],
+                'encrypt': int(data['encryption']),
+                'encryptkey': enc_key,
+            }
+            pool_id = await self.middleware.call(
+                'datastore.insert',
+                'storage.volume',
+                pool,
+                {'prefix': 'vol_'},
+            )
+
+            await self.__save_encrypteddisks(pool_id, enc_disks, disks_cache)
+
+            await self.middleware.call(
+                'datastore.insert',
+                'storage.scrub',
+                {'volume': pool_id},
+                {'prefix': 'scrub_'},
+            )
+        except Exception as e:
+            # Something wrong happened, we need to rollback and destroy pool.
+            try:
+                await self.middleware.call('zfs.pool.delete', data['name'])
+            except Exception:
+                self.logger.warn('Failed to delete pool on pool.create rollback', exc_info=True)
+            if pool_id:
+                await self.middleware.call('datastore.delete', 'storage.volume', pool_id)
+            raise e
+
+        # There is really no point in waiting all these services to reload so do them
+        # in background.
+        async def restart_services():
+            await self.middleware.call('service.reload', 'disk')
+            await self.middleware.call('service.start', 'ix-syslogd')
+            await self.middleware.call('service.restart', 'system_datasets')
+            # regenerate crontab because of scrub
+            await self.middleware.call('service.restart', 'cron')
+            # restart smartd to enable monitoring for any new drives added
```

Comment:
I don't understand how does creating pool leads to adding new drives, am I missing something?

Reply:
Thats a great question, I have simply copied existing code to middlewared. I cant think of a reason this was done, I will evaluate and remove.